common/src/main/java/io/pravega/common/hash/HashHelper.java

```diff
@@ -160,4 +167,77 @@ public int getAsInt() {
             return getHashCode().asInt();
         }
     }
+    
+    /**
+     * Hashes a bufferview in-place. (in contrast to murmur above which needs to copy data into a single array)
+     * (This passes the smhasher quality test suite)
+     *
+     * @param bufferView The input.
+     */
+    public static final long hashBufferView(BufferView bufferView) {
+        long multiple = 6364136223846793005L; // From knuth
+        long inc = 1442695040888963407L; // From knuth
+        long state = 0xc3a5c85c97cb3127L; // (arbitrary, from farmhash) 
+        long wheil = 0x9ae16a3b2f90404fL; // (arbitrary, from farmhash) 
+        ByteBuffer leftOvers = ByteBuffer.wrap(new byte[16]);
+        val iter = bufferView.iterateBuffers();
+        while (iter.hasNext()) {
+            ByteBuffer buffer = iter.next();
+            if (leftOvers.position() > 0) {
+                ByteBufferUtils.copy(buffer, leftOvers);
+                if (!leftOvers.hasRemaining()) {
+                    leftOvers.flip(); 
+                    state = updateHashState(state, wheil, leftOvers);
+                    wheil += inc; 
+                    leftOvers.clear();
+                }
+            }
+            while (buffer.remaining() >= 16) {
+                state = updateHashState(state, wheil, buffer);
+                wheil += inc;
+            }
+            if (buffer.hasRemaining()) {
+                ByteBufferUtils.copy(buffer, leftOvers);
+            }
+        }
+        leftOvers.flip();
+        while (leftOvers.hasRemaining()) {
+            byte b = leftOvers.get();
+            state = (state ^ b) * wheil;
+            state ^= state >> 47;
+            wheil += inc;
+        }
+        int rot = (int) state & 63; //RR permutation from PCG
```

Comment:
magic number

Reply:
Um, ok. I can make a constant but it just seems like it would make the code worse. Like what would you even name it? "SIXTY_THREE"? "ONE_LESS_THAN_THE_NUMBER_OF_BITS_IN_A_LONG"?